pkg/validate/validate_component_test.go

```diff
@@ -15,405 +15,249 @@
-  spec:
-    componentName: foo-comp
-    version: 1.0.2
-- apiVersion: 'bundle.gke.io/v1alpha1'
-  kind: Component
-  metadata:
-    name: foo-comp-1.0.2-zed
-  spec:
-    componentName: foo-comp
-    version: 1.0.2`,
-			errSubstring: "component reference",
-		},
-
-		{
-			desc: "fail: component invalid X.Y.Z version string ",
-			set:  defaultComponentSetNoRefs,
-			components: `
-components:
-- apiVersion: 'bundle.gke.io/v1alpha1'
-  kind: Component
-  metadata:
-    name: foo-comp-1.0.2
-  spec:
-    componentName: foo-comp
-    version: 2.010.1`,
-			errSubstring: "must be of the form X.Y.Z",
-		},
-		{
-			desc: "fail: component invalid X.Y.Z app version string ",
-			set:  defaultComponentSetNoRefs,
-			components: `
-components:
-- apiVersion: 'bundle.gke.io/v1alpha1'
-  kind: Component
-  metadata:
-    name: foo-comp-1.0.2
-  spec:
-    componentName: foo-comp
-    version: 2.10.1,
-    appVersion: 2.010.1`,
-			errSubstring: "must be of the form X.Y.Z",
-		},
-		{
-			desc: "fail: component missing X.Y.Z version string ",
-			set:  defaultComponentSetNoRefs,
-			components: `
-components:
-- apiVersion: 'bundle.gke.io/v1alpha1'
-  kind: Component
-  metadata:
-    name: foo-comp-1.0.2
-  spec:
-    componentName: foo-comp`,
-			errSubstring: "Required value",
-		},
+func assertValidateComponent(componentString string, numErrors int, t *testing.T) {
+  component, errParse := converter.FromYAMLString(componentString).ToComponent()
+  if errParse != nil {
+    t.Fatalf("parse error: %v", errParse)
+  }
+  errs := Component(component)
+  numErrorsActual := len(errs)
+  if numErrorsActual != numErrors {
+    t.Fatalf("expected %v errors got %v,  errors: %v", numErrors, numErrorsActual, errs)
+  }
+}
 
-		{
-			desc: "component object success",
-			set:  defaultComponentSet,
-			components: `
-components:
-- apiVersion: 'bundle.gke.io/v1alpha1'
-  kind: Component
-  metadata:
-    name: foo-comp-1.0.2
-  spec:
-    componentName: foo-comp
-    version: 1.0.2
-    objects:
-    - apiVersion: v1
-      kind: Pod
-      metadata:
-        name: foo-pod`,
-		},
-		{
-			desc: "object fail: duplicate",
-			set:  defaultComponentSet,
-			components: `
-components:
-- apiVersion: 'bundle.gke.io/v1alpha1'
-  kind: Component
-  metadata:
-    name: foo-comp-1.0.2
-  spec:
-    componentName: foo-comp
-    version: 1.0.2
-    objects:
-    - apiVersion: v1
-      kind: Pod
-      metadata:
-        name: foo-pod
-    - apiVersion: v1
-      kind: Pod
-      metadata:
-        name: foo-pod`,
-			errSubstring: "object reference",
-		},
-		{
-			desc: "object fail: no metadata.name",
-			set:  defaultComponentSet,
-			components: `
-components:
-- apiVersion: 'bundle.gke.io/v1alpha1'
-  kind: Component
-  metadata:
-    name: foo-comp-1.0.2
-  spec:
-    componentName: foo-comp
-    version: 1.0.2
-    objects:
-    - apiVersion: v1
-      kind: Pod`,
-			errSubstring: "Required value",
-		},
-	}
+func TestValidateComponents(t *testing.T) {
```

Comment:
You removed all the err substring stuff. Why? I think that's very valuable to ensure you're actually getting the errors that you want.

Reply:
added it back